### ISSUE TITLE
Subclasses of OrderedDict are not serialized correctly.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#1803]
See discussion in #1800.

I tried registering a custom serializer, but it didn't help when the object was created on a worker.

**Comment by andyljones** [comment]
Looks like it's a problem with PyArrow. Minimal Ray example:

```python
import ray
from collections import OrderedDict
ray.init()

class SubDict(OrderedDict):
    pass

@ray.remote
def f():
    return SubDict()

y = ray.get(f.remote())
type(y)
```

Minimal PyArrow example:

```python
import pyarrow
import collections

class SubDict(OrderedDict):
    pass

type(pyarrow.serialize(SubDict()).deserialize())
```

My spidey sense says that this is going to be caused by the custom serialisation of OrderedDicts [built into PyArrow](https://github.com/apache/arrow/pull/1152) being too promiscuous. I'll have a look later, but since I don't fancy diving into PyArrow internals I'm probably going to solve my immediate problem by just not subclassing OrderedDicts.

**Comment by robertnishihara** [comment]
You're right that it's a pyarrow issue and related to the issue in https://github.com/ray-project/ray/issues/4463.

Another workaround could be to simply register a custom serializer for your class (see https://ray.readthedocs.io/en/latest/api.html#ray.register_custom_serializer), e.g., something like
